crates/eigenda/src/eigenda_data.rs

```diff
@@ -69,4 +126,39 @@ mod tests {
         assert!(result.is_err());
         assert_eq!(result.unwrap_err(), BlobDecodingError::InvalidLength);
     }
+
+    #[test]
+    fn test_encode_and_decode_success() {
+        let rollup_data = vec![1, 2, 3, 4];
+        let eigenda_blob = EigenDABlobData::encode(&rollup_data);
+        let data_len = eigenda_blob.blob.len();
+        assert!(data_len.is_power_of_two());
```

Comment:
remove the old tests that use the custom generate_blob_data function; also delete that function now that we have encode